homework_3week.js

```diff
@@ -0,0 +1,46 @@
+/* Написать свою реализацию методов map, filter которые будут полностью
+ повторять функционал существующих. Данные функции записать в прототип.
+
+Пример:
+Array.prototype.customMap = function () {
+  // your implementation here
+}
+*/
+arr = [1,2,3,4,5];
+
+Array.prototype.customMap = function (callback) {
+    if(!Array.isArray(arr)) return console.log('Ошибка');
+
+    const newArray = []
+    for(let index = 0; index < this.length ; index ++){
+        const valueCallback = callback(this[index],index);
+        newArray.push(valueCallback);
+    }
+    return newArray;
+}
+
+const result = arr.customMap((item,index)=>{
+    return item * 3;
+})
+console.log(result);
+
+
+// теперь для метода Filter:
+
+Array.prototype.customFilter = function (callback) {
+    if(!Array.isArray(arr)) return console.log('Error');
```

Comment:
from where you have taken `arr` variable? if you try to run this code you will receive an error. `arr is not defined`